catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImplTest.java

```diff
@@ -13,6 +13,1727 @@
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithInvalidMetacardNoRegistryTag() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithIngestException() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        when(catalogFramework.create(any(CreateRequest.class))).thenThrow(IngestException.class);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test(expected = FederationAdminException.class)
+    public void testAddRegistryEntryWithSourceUnavailableException() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String destination = TEST_DESTINATION;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        Set<String> destinations = new HashSet<>();
+        destinations.add(destination);
+
+        when(catalogFramework.create(any(CreateRequest.class))).thenThrow(SourceUnavailableException.class);
+
+        federationAdminServiceImpl.addRegistryEntry(metacard, destinations);
+
+        verify(catalogFramework, never()).create(any(CreateRequest.class));
+    }
+
+    @Test
+    public void testAddRegistryEntryMetacard() throws Exception {
+        String registryId = RegistryObjectMetacardType.REGISTRY_ID;
+        String metacardId = TEST_METACARD_ID;
+        Metacard metacard = getTestMetacard();
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                registryId));
+
+        metacard.setAttribute(new AttributeImpl(Metacard.TAGS,
+                Collections.singletonList(RegistryConstants.REGISTRY_TAG)));
+
+        Metacard createdMetacard = metacard;
+        createdMetacard.setAttribute(new AttributeImpl(Metacard.ID, metacardId));
+
+        Subject systemSubject = security.getSystemSubject();
```

Comment:
There also seems to be a lot of duplication of mocking a catalog create response.  Consider extracting a helper method.